src/scripts/install_machine.py

```diff
@@ -0,0 +1,334 @@
+
+                params.append(f"user-data={file.name}")
+
+                files.append(file.name)
+
+
+def prepare_installation_source(params, args):
+    only_define = False
+    if args.get('type') == 'create':
+
+        if not args.get('startVm'):
+            only_define = True
+            params.append("--print-xml")
+
+    if args.get('sourceType') == "pxe":
+        params += ['--pxe', '--network', args.get('source')]
+    elif args.get('sourceType') == "os":
+        params += ['--install', f"os={args.get('os')}"]
+    elif args.get('sourceType') == "disk_image" or args.get('sourceType') == 'cloud':
+        params.append("--import")
+    elif (args.get('source')[0] == '/' and os.path.isfile(args.get('source'))) or (args.get('sourceType') == 'url' and args.get('source').endswith(".iso")):
+        if only_define:
+            return
+        params += ['--cdrom', args.get('source')]
+    else:
+        if only_define:
+            return
+        params += ['--location', args.get('source')]
+
+
+def prepare_virt_install_params(args, files):
+    logging.debug(args)
+
+    params = [
+        "virt-install",
+        "--connect", f"qemu:///{args.get('connectionName')}",
+        "--quiet",
+        "--name", args.get('vmName'),
+        "--os-variant", args.get('os'),
+        "--memory", str(args.get('memorySize')),
+    ]
+
+    if args.get('type') == 'install' or (args.get('storagePool') and args.get('storagePool') != 'NewVolume'):
+        params += ["--check", "path_in_use=off"]
+
+    if args.get('sourceType') != 'disk_image':
+        params += ["--wait", "-1"]
+
+    if args.get('type') == 'install' or (args.get('startVm') and args.get('sourceType') != 'disk_image'):
+        params.append("--noautoconsole")
+
+    # Disks
+    if args.get('type') == 'install':
+        has_param = False
+        for disk in args.get('disks'):
+            params += ['--disk', disk]
+            has_param = True
+
+        if not has_param:
+            params += ['--disk', "none"]
+    else:
+        params.append("--disk")
+        if args.get('sourceType') == "cloud":
+            backing_store = f",backing_store={args.get('source')},"
+        else:
+            backing_store = ""
+
+        if args.get('sourceType') == 'disk_image':
+            params.append(f"{args.get('source')},device=disk")
+        elif args.get('storagePool') == 'NoStorage':
+            params.append("none")
+        elif args.get('storagePool') != 'NewVolume':
+            params.append(f"vol={args.get('storagePool')}/{args.get('storageVolume')}{backing_store}")
+        else:
+            params.append(f"size={args.get('storageSize')},format=qcow2{backing_store}")
+
+    # NICs
+    if args.get('type') == "install":
+        has_param = False
+        for network in args.get('vnics'):
+            params += ['--network', network]
+            has_param = True
+
+        if not has_param:
+            params += ['--network', "none"]
```

Comment:
stylistic suggestion
```suggestion
        vnics = args['vnics'] or ['none']
        for network in vnics:
            params += ['--network', network]
```

...and similar for the graphics stuff below